### PR TITLE
Stronger validation

### PR DESCRIPTION
**Pull request by brisvag**
First of all, sorry for the big blob of changes, at this stage it's aways easier to refactor a lot of things than splitting in small atomic changes... In summary:

validation was not quite there yet; especially ndim guessing was failing regularly (`reshape(-1, 2)` was working for any array with even number of elements :P). I added a bunch of checks for data types and values that hopefully should ensure the object is indeed complying.

In here, we currently intermix 2 types of validation without distinction, which is probably bad: *checking* if thigns are correct (`validate_cryopose`) and *enforcing* correct things as much as possible (`validate_positions`). We should probably formalize this, because I can see cases where someone wants the positions in 3D no matter what (just pad it if 2D)

### CODE REVIEW

File: cryopose/_validators.py [hunk 1]
```diff
@@ -1,24 +1,53 @@
...
+    if ndim != positions.shape[1]:
+        raise ValueError(f'positions are {positions.shape[1]}D, but a {ndim}D cryopose was requested')
+    return positions
+
+
+def validate_orientations(orientations: Union[Rotation, Sequence[Rotation]], ndim: Literal[2, 3] = 3) -> Rotation:
+    orientations = Rotation.concatenate(orientations)
+    if ndim == 2:
+        # TODO: check that rotation is only inplane?
+        ...
```
Comment (alisterburt) [inline cryopose/_validators.py:25]: 2D particles can still have 3D rotations :)

Need to decide on semantics for what a 3D orientation for a 2D particle means

**Review by alisterburt — APPROVED** [review]
Looks good, merging so we can iterate from here

File: cryopose/_validators.py (hunk 1 above)
Comment (alisterburt) [reply cryopose/_validators.py:25]: 3D tomogram space doesn't really exist for single particle so the semantics can't be exactly the same :)

**Comment by brisvag** [comment]
Objeying to pre-commit made me do some more changes... weirdly, pre-commit *really insisted on 2 things:
- min python version should be 3.7, not 3.8
- `Literal` should be hidden behind `if TYPE_CHECKING`

No clue why.

**Review by alisterburt — APPROVED** [review]
Looks good!

**Pull request merged**